app/electron/main.js

```diff
@@ -123,11 +123,37 @@ const isDev = process.mainModule.filename.indexOf('app.asar') === -1 ||
     process.mainModule.filename.indexOf('app') === -1;
 
 let serverProcess = null;
+let intentionalQuit = false;
+let serverProcessQuit = false;
+
 function createWindow () {
   let serverFilePath;
   if (!isDev) {
     serverFilePath = path.join(process.resourcesPath, './electron/server');
     serverProcess = spawn(serverFilePath, {shell: true});
+
+    serverProcess.stdout.on('data', (data) => {
+      log.info(`server process stdout: ${data}`);
+    });
+    serverProcess.stderr.on('data', (data) => {
+      const sterrMessage = `server process stderr: ${data}`;
+      if (data && data.indexOf && data.indexOf("Requesting") !== -1) {
+        // The server prints out urls it's getting, which aren't errors.
+        log.info(sterrMessage);
+      } else {
+        log.error(sterrMessage);
+      }
+    });
+    serverProcess.on('close', (code, signal) => {
+      const closeMessage = `server process process exited with code:${code} signal:${signal}`;
+      if (!intentionalQuit) {
+        //TODO: message mainWindow, or loadURL to an error url?
+        log.error(closeMessage);
+      } else {
+        log.info(closeMessage);
+      }
+      serverProcessQuit = true;
+    });
```

Comment:
@illume can we move these event handlers on serverProcess into a seperate function maybe something like ```attachServerEventHandlers```(which resides in the global scope of main.js) because otherwise than in the headless mode feature  https://github.com/kinvolk/headlamp/pull/154 I will have to copy the entire server handlers again

Reply:
Good idea. Thanks @ashu8912 

I pushed a commit for this.